audiostream/src/ni/media/audio/os/win/media_foundation_helper.h

```diff
@@ -86,7 +86,8 @@ class SyncronousByteStream : public IMFByteStream
 
         try
         {
-            *read = static_cast<ULONG>( m_source.read( reinterpret_cast<char*>( buffer ), toRead ) );
+            const auto result =  m_source.read( reinterpret_cast<char*>( buffer ), toRead );
+            *read = static_cast<ULONG>( result == -1 ? 0 : result );
```

Comment:
would it make sense to define -1 as "endOfSequence" for readability purposes?

Reply:
Probably better to avoid magic numbers indeed.